### ISSUE TITLE
Cant run "wine powershell"

### ISSUE THREAD

**Issue by niculw** [PietJankbal/powershell-wrapper-for-wine#1]
When running `wine powershell` i get the following

```
Downloading File From: https://github.com/PowerShell/PowerShell/releases/download/v7.0.3/PowerShell-7.0.3-win-x64.msi, To: PowerShell-7.0.3-win-x64.msi 
Something went wrong :( (32-bit?, winversion <win7?, .... 
```
Yet im using a 64bit prefix set to windows 7, in wine 5.0
What wine are you using?

**Comment by PietJankbal** [comment]
Hi,

That`s probably because the installer is failing because a stub pwrshplugin.dll is missing. It was only added in wine-5.11

I think you can work-around it in wine-5.0 by just copy some dll with high enough version resource to (a fake)  pwrshplugin.dll.
Assuming ~/.wine is your prefix:

cp ~/.wine/drive_c/windows/system32/msxml3.dll ~/.wine/drive_c/windows/system32/pwrshplugin.dll
cp ~/.wine/drive_c/windows/syswow64/msxml3.dll ~/.wine/drive_c/windows/syswow64/pwrshplugin.dll
 
Side note: The download of the installer is currently terribly slow. I `m trying to find another downloadlink

**Comment by niculw** [comment]
Updated to wine-5.18 and did as you wrote. Now i get:

```
0034:err:menubuilder:convert_to_native_icon error 0x80004005 getting frame 1
0034:err:menubuilder:convert_to_native_icon error 0x88982F04 committing encoder
0034:err:menubuilder:convert_to_native_icon error 0x80004005 getting frame 2
0034:err:menubuilder:convert_to_native_icon error 0x88982F04 committing encoder
Downloading File From: https://github.com/PowerShell/PowerShell/releases/downloa
d/v7.0.3/PowerShell-7.0.3-win-x64.msi, To: PowerShell-7.0.3-win-x64.msi
0024:fixme:ntdll:EtwRegisterTraceGuidsA (7127880F, 712B9A98, {0cfe0455-93ba-440d-a3fe-553973d0b723}, 1, 0060E24C, (null), (null), 712B9AA0): stub
0024:fixme:ntdll:EtwRegisterTraceGuidsA (7127880F, 712B9AB8, {797fabac-7b58-4796-b924-d51178a59ce4}, 1, 0060E24C, (null), (null), 712B9AC0): stub
0024:fixme:ntdll:EtwEventRegister ({43d1a55c-76d6-4f7e-995c-64c711e5cafe}, 7120A8A5, 712BB318, 712BB110) stub.
0024:fixme:shell:SHQueueUserWorkItem Unsupported arguments
0024:fixme:ras:RasEnumEntriesW (00000000,(null),001D3160,0060E34C,001D304C),stub!
00e8:fixme:ras:RasConnectionNotificationW (FFFFFFFF,000000FC,0x00000003),stub!
0024:fixme:ntdll:EtwEventActivityIdControl 0x1, 0060E604: stub
00e8:fixme:shell:SHQueueUserWorkItem Unsupported arguments
00e8:fixme:ras:RasEnumEntriesW (00000000,(null),001ED3C8,00C6FD00,001ED2B4),stub!
00e8:fixme:winsock:WSALookupServiceBeginW (0xc6fd48 0x00000210 0x1d2250) Stub!
00f4:fixme:ras:RasConnectionNotificationW (FFFFFFFF,000000FC,0x00000003),stub!
00e8:fixme:shell:SHQueueUserWorkItem Unsupported arguments
00f4:fixme:ras:RasConnectionNotificationW (FFFFFFFF,000000FC,0x00000003),stub!
00e8:fixme:ras:RasEnumEntriesW (00000000,(null),001EDE00,00C6F964,001EDCEC),stub!
00e8:fixme:shell:SHQueueUserWorkItem Unsupported arguments
00f4:fixme:ras:RasConnectionNotificationW (FFFFFFFF,000000FC,0x00000003),stub!
00e8:fixme:ras:RasEnumEntriesW (00000000,(null),001EE2F8,00C6F6E0,001EE1E4),stub!
00e8:fixme:winsock:server_ioctl_sock Unsupported ioctl 9800012c (device=9800 access=0 func=4b method=0)
00e8:fixme:winsock:WSAIoctl unsupported WS_IOCTL cmd (SIO_SET_COMPATIBILITY_MODE)
00f4:fixme:crypt:CRYPT_RegControl CERT_STORE_CTRL_AUTO_RESYNC: stub
00f4:fixme:crypt:CRYPT_RegControl CERT_STORE_CTRL_AUTO_RESYNC: stub
Something went wrong :( (32-bit?, winversion <win7?, ....
```
So still alot of errors

**Comment by PietJankbal** [comment]
Hi,

Sorry I can`t reproduce that output. Is this in a clean WINEPREFIX, or did you install other things in it? 
In fact after the message "Downloading File" there should some fixme`s from urlmon and wininet printed; This is what i see:

Downloading File From: https://github.com/PowerShell/PowerShell/releases/downloa
d/v7.0.3/PowerShell-7.0.3-win-x64.msi, To: PowerShell-7.0.3-win-x64.msi
0024:fixme:urlmon:InternetBindInfo_GetBindString not supported string type 20
00e4:fixme:wininet:set_cookie secure not handled
00e4:fixme:wininet:set_cookie Unknown additional option L"SameSite=Lax"

I don`t see that in your output. So are you using a native urlmon/wininet?

I cannot really test atm as the download seems so slow, don`t know why. When try manually download:

wget https://github.com/PowerShell/PowerShell/releases/download/v7.0.3/PowerShell-7.0.3-win-x64.msi

it takes more than half an hour; Usually it should finish in a few seconds. I don`t know why this is. Maybe that`s part of the problem?????

Regards

**Comment by niculw** [comment]
I cleaned the wineprefix and re did the installation. now it worked. the download was fairly quick.

**Comment by niculw** [comment]
However im still having issues installing Waves Central using this. It still errors out saying it cant use powershell.

**Comment by PietJankbal** [comment]
Try cd into directory where the exe is:

cd ~/.wine/drive_c/Program\ Files/Waves\ Central/ or something the like,

and then try run again wine Waves Central.exe (use exact name bunary)

**Comment by niculw** [comment]
Got me further, yet Waves Central opens to a black screen, it calls an error about urlmon (disappears when installing urlmon via winetricks), The remaining error is:
```
0x000000014287009f EntryPoint+0xfdb56c3f in waves central: int	$3
Modules:
Module	Address					Debug info	Name (68 modules)
PE	          220000-          455000	Deferred        setupapi
PE	          460000-          610000	Deferred        winmm
PE	          620000-          92a000	Deferred        d3d11
PE	          930000-          bd1000	Deferred        comdlg32
PE	         13a0000-         13c6000	Deferred        powrprof
PE	        61540000-        61662000	Deferred        advapi32
PE	        61d40000-        61d5f000	Deferred        uiautomationcore
PE	        62980000-        6298f000	Deferred        api-ms-win-core-localization-l1-2-1
PE	        62cc0000-        62cde000	Deferred        ncrypt
PE	        62dc0000-        6306b000	Deferred        rpcrt4
PE	        63280000-        632a0000	Deferred        version
PE	        635c0000-        636f6000	Deferred        oleacc
PE	        639c0000-        63a12000	Deferred        shcore
PE	        64840000-        6498f000	Deferred        shlwapi
PE	        64cc0000-        651cf000	Deferred        oleaut32
PE	        65380000-        653a7000	Deferred        wtsapi32
PE	        664c0000-        664f2000	Deferred        hid
PE	        66c40000-        66c4f000	Deferred        api-ms-win-appmodel-runtime-l1-1-2
PE	        67600000-        6760e000	Deferred        api-ms-win-core-string-l1-1-0
PE	        67a80000-        67c28000	Deferred        d3d9
PE	        67e80000-        67ecc000	Deferred        jsproxy
PE	        68300000-        68490000	Deferred        combase
PE	        68500000-        685b1000	Deferred        uxtheme
PE	        69400000-        69542000	Deferred        winhttp
PE	        69740000-        697c1000	Deferred        propsys
PE	        69e40000-        69eb9000	Deferred        msacm32
PE	        6a100000-        6a12a000	Deferred        userenv
PE	        6a200000-        6a7ba000	Deferred        ole32
PE	        6a840000-        6a95b000	Deferred        usp10
PE	        6aa80000-        6aa8e000	Deferred        api-ms-win-core-fibers-l1-1-1
PE	        6b300000-        6b30e000	Deferred        api-ms-win-core-localization-obsolete-l1-2-0
PE	        6b7c0000-        6b7db000	Deferred        dhcpcsvc
PE	        6ba00000-        6baca000	Deferred        sechost
PE	        6ca40000-        6cf91000	Deferred        comctl32
PE	        6dc80000-        6dc8f000	Deferred        api-ms-win-core-synch-l1-2-0
PE	        6ea40000-        6ea60000	Deferred        dwmapi
PE	        6eb00000-        6f2ae000	Deferred        user32
PE	        6f880000-        6fb3b000	Deferred        dbghelp
PE	        6fd80000-        6fd8e000	Deferred        api-ms-win-core-datetime-l1-1-1
PE	        70c80000-        70c97000	Deferred        msimg32
PE	        71000000-        71058000	Deferred        imm32
PE	        710c0000-        7111b000	Deferred        dxva2
PE	        7a850000-        7a854000	Deferred        opengl32
PE	        7b000000-        7b3e5000	Deferred        kernelbase
PE	        7b610000-        7b7cd000	Deferred        kernel32
PE	        7bc00000-        7bf88000	Deferred        ntdll
PE	       140000000-       1466df000	Export          waves central
PE	       180000000-       180369000	Deferred        ffmpeg
PE	     7ff776d0000-     7ff777fa000	Deferred        wininet
PE	     7ff7c5a0000-     7ff7c7f9000	Deferred        iertutil
PE	     7ff7c800000-     7ff7c978000	Deferred        urlmon
PE	    7fc97d680000-    7fc97d683000	Deferred        kerberos
PE	    7fc97dac0000-    7fc97dac4000	Deferred        winex11
PE	    7fc97ddd0000-    7fc97ddd4000	Deferred        msvcrt
PE	    7fc97dfe0000-    7fc97dfe4000	Deferred        dnsapi
PE	    7fc97e000000-    7fc97e003000	Deferred        netapi32
PE	    7fc97e050000-    7fc97e053000	Deferred        secur32
PE	    7fc97e0a0000-    7fc97e0ab000	Deferred        winspool
PE	    7fc97e100000-    7fc97e103000	Deferred        dwrite
PE	    7fc97e1d0000-    7fc97e1d4000	Deferred        wined3d
PE	    7fc97e370000-    7fc97e374000	Deferred        dxgi
PE	    7fc97e3c0000-    7fc97e3c4000	Deferred        iphlpapi
PE	    7fc97e3f0000-    7fc97e3f4000	Deferred        bcrypt
PE	    7fc97e420000-    7fc97e464000	Deferred        crypt32
PE	    7fc97e520000-    7fc97e524000	Deferred        ws2_32
PE	    7fc97e580000-    7fc97ee57000	Deferred        shell32
PE	    7fc97efd0000-    7fc97efd7000	Deferred        gdi32
PE	    7fc97f290000-    7fc97f294000	Deferred        ucrtbase
Threads:
process  tid      prio (all id:s are in hex)
00000038 services.exe
	0000003c    0
	00000040    0
	0000004c    0
	0000006c    0
	00000084    0
	000000bc    0
00000044 winedevice.exe
	00000048    0
	00000054    0
	00000058    0
	000000a4    0
0000005c plugplay.exe
	00000060    0
	00000070    0
	00000074    0
	00000078    0
	00000094    0
	00000098    0
	000001a4    0
00000064 explorer.exe
	00000068    0
	000000a8    0
	000000ac    0
0000007c winedevice.exe
	00000080    0
	00000088    0
	0000008c    0
	00000090    0
	0000009c    0
	000000a0    0
000000b4 rpcss.exe
	000000b8    0
	000000c4    0
	000000c8    0
	000000cc    0
	000000d0    0
	000000d4    0
000000dc Waves Central.exe
	000000e0    0
	000000ec    0
	000000f0    0
	000000f4    0
	000000f8    0
	000000fc   -2
	00000100    0
	00000104    0
	00000108    0
	0000010c    0
	00000110    0
	00000114    0
	00000118    0
	0000011c    0
	00000120    0
	00000124    0
	00000128    0
	0000012c    0
	00000130    0
	00000134    0
	00000138    0
	0000013c    0
	00000140    0
	0000014c    0
	00000150    0
	00000154    0
	00000158    0
	0000015c    0
	00000160    0
	00000164    0
	00000168    0
	00000198    0
	0000019c    0
	000001a0    0
	000001a8    0
	000001b4    0
	00000250   -2
	00000254   -2
	00000258   -2
	00000344    0
	00000348    0
	0000034c    0
	00000350    0
	0000035c    0
	00000360    0
	00000364    0
	0000036c    0
	00000510    0
	00000514    0
	00000518    0
000000e4 conhost.exe
	000000e8    0
00000144 Waves Central.exe
	00000148    1
	0000016c    0
	00000170    0
	00000174    0
	00000178    0
	0000017c   -2
	00000180    1
	00000184    0
	00000188    1
	0000018c    0
	00000190    0
	00000194    0
	00000370    0
	0000037c    0
0000025c Waves Central.exe
	00000260    0
	00000274    0
	00000278    0
	0000027c    0
	00000280    0
	00000284   -2
	00000288    0
	0000028c    0
	00000290    0
	00000294    0
	00000298    0
	0000029c    0
	000002a0    0
	000002a4    0
	00000368   -2
00000354 powershell.exe
	00000358    0
00000374 (D) C:\Program Files\Waves Central\Waves Central.exe
	00000378    0 <==
	000003b0    0
	000003b4    0
	000003b8    0
	000003bc    0
	000003c0   -2
	000003c4    1
	000003c8    0
	000003cc    0
	000003d0    1
	000003d4    0
	000003d8    0
	000003dc    0
	000003e0    0
	000003e4    0
	000003e8   -2
	000003ec    0
	000003f0    0
	000003f4    0
	000003f8    0
	0000041c    0
	00000430    0
	00000434    0
	0000050c    0
00000394 cmd.exe
	00000398    0
000003a8 ConEmu.exe
	000003ac    1
	000003fc    0
	00000400    0
	00000450    0
	0000045c    0
	00000460    0
	000004fc    0
	00000500    0
00000464 ConEmuC64.exe
	00000468    0
	000004f0    1
	000004f4    1
	000004f8    1
	00000504    0
	00000508    0
	0000053c    0
0000046c wineconsole.exe
	00000470    0
00000534 pwsh.exe
	00000538    0
	00000540    0
	00000544    2
	00000548    0
	0000054c    0
	00000554    0
	0000055c    0
	00000560    0
	00000564    0
	00000570    0
	00000574    0
	00000578    0
	0000057c    0
	00000580    0
	00000584    0
	00000588    0
	0000058c    0
	00000590    0
	00000594    0
	00000598    0
	0000059c    0
	000005a0    0
	000005a4    0
	000005a8    0
	000005ac    0
	000005b0    0
	000005b8    0
System information:
    Wine build: wine-5.18
    Platform: x86_64
    Version: Windows 7
    Host system: Linux
    Host version: 5.8.13-050813-generic
```

**Comment by PietJankbal** [comment]
Hi,

I guess you need "winetricks arial " (no need for urlmon btw afaik, consider setting it back to builtin)

(If black screen remains try disabling libglesv2.dll)

**Comment by niculw** [comment]
I installed arial and recieved no more errors. I then tried `WINEDLLOVERRIDES=libglesv2.dll=d wine Waves\ Central.exe` and can confirm it works. I set urlmon back to native btw.

So how to install waves central 11 in wine

Get powershell using your scripts on a clean prefix,
Install it

Set up environment for Waves Central according to the guide: https://blog.inagaki.in/en/post/linux-waves-plugins/
I think??? remember arial!
When installing "Install_Waves_Central.exe" let it error out.
Go to .wine/drive_c/Program Files/Waves Central
run `WINEDLLOVERRIDES=libglesv2.dll=d wine Waves\ Central.exe`

Agreed or im missing something?

**Comment by PietJankbal** [comment]
>Agreed or im missing something?

Yeah, I guess you didn`t miss anything, but note that the "winetricks arial" and "disable libglesv2" might not be necessary for everyone. Eg. I only needed the "winetricks arial", but if one uses Wine-Staging it might not be necessary (though it won`t hurt either...)

As for the disabling of libglesv2: I didn`t need it, I think it depends on what graphics card/driver you have. Also i think "winetricks dxvk151" (or other dxvkxxx) could work around it. Or passing command line option like --disable-gpu, but didn`t try that.
A lot of info on black screen bug is at https://bugs.winehq.org/show_bug.cgi?id=44985

Last note: If you encounter any bugs in Waves Central due to failing powershell commands, you can hopefully (no guarantee :) ) work around it by adjusting the source in main.c (and recompile) ; Would be nice if you posted failing commands here.

Regards

**Comment by PietJankbal** [comment]
Note: The error it ran into when windowsversion set to win10 should be fixed in most recent Staging version (6.0-rc3).
Regards
